avm/ptn/app-service-lza/hosting-environment/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,51 @@
+metadata name = 'Using only defaults.'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'applzamin'
+
+@description('Optional. Test name prefix.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
```

Comment:
Unless not possible, this deployment should use the syntax to test idempotency.

Reply:
I take it it's not possible as you resolved the comment? 

On that note, please only resolve comments that are 'code suggestions' and just comment on the others once addressed. It makes the re-review a lot easier for the reviewer :)

Reply:
I think I interpreted the comment as reducing the number of parameters that need to be provided that don't have to defaults to test that it deploys the same configuration each time. Are you referring to the auto generated password each time? Sorry I prob should've asked for clarification before.

Reply:
Pardon. We can maybe also connect some time offline.

What you're referring to was the correct reponse to a different comment 😄 . It is correct that the `defaults` test should test your defaults by you only providing the minimum required parameters.

**Independent** of that (and what this comment was referring to) is something you can see here:
https://github.com/Azure/bicep-registry-modules/blob/ff8dc79cd4cc2794692fee8d6ec7b4e51b36e572/avm/res/key-vault/vault/tests/e2e/defaults/main.test.bicep#L40

Effectively to run the deployment in a loop that validates that the resource can be deployed in an idempotent way - that is - 2+ times in a row without falling over itself (which **should** be the norm for all resource - but as usual has exceptions to it)